src/MobileApps/MyDriving/MyDriving.UWP/Views/CurrentTripView.xaml.cs

```diff
@@ -267,22 +267,22 @@ private void ClearExtendedExecution()
             {
                 // Need to update Map UI before we start saving. So that the entire trip is visible. 
                 UpdateMap_PositionChanged(basicGeoposition);
+               
+                // Changing the record button icon and andding end marker earlier to notify user about the stop process. 
+                recordButtonImage = new BitmapImage(new Uri("ms-appx:///Assets/StartRecord.png", UriKind.Absolute));
+                OnPropertyChanged(nameof(RecordButtonImage));
+                AddEndMarker(basicGeoposition);
 
                 if (!await ViewModel.StopRecordingTrip())
                 {
                     recordButtonIsBusy = false;
                     return;
                 }
 
-                // Need to add the end marker only when we are able to stop the trip. 
-                AddEndMarker(basicGeoposition);
+                await ViewModel.SaveRecordingTripAsync();
```

Comment:
The previous comment was based on my assumption that we needed to update end marker only once we could save successfully. I think, the end marker should be put if user can stop and since  users have complained about this, I made this change to move it before Stopping the trip. The previous comment did not make sense now.